src/SamplesApp/SamplesApp.UITests/TestFramework/ImageAssert.cs

```diff
@@ -235,4 +279,51 @@ public static void DoesNotHaveColorAt(FileInfo screenshot, float x, float y, Col
 		private static string ToArgbCode(Color color)
 			=> $"{color.A:X2}{color.R:X2}{color.G:X2}{color.B:X2}";
 	}
+
+	public struct ExpectedPixels
+	{
+		public static ExpectedPixels At(string name, float x, float y)
+			=> new ExpectedPixels(name, x, y, new Color[0, 0]);
+
+		public static ExpectedPixels At(float x, float y, [CallerLineNumber] int line = -1)
+			=> new ExpectedPixels($"at line: {line}", x, y, new Color[0,0]);
+
+		public ExpectedPixels Pixels(string[,] pixels) => new ExpectedPixels(Name, X, Y, pixels, Tolerance);
+
+		public ExpectedPixels WithTolerance(byte tolerance) => new ExpectedPixels(Name, X, Y, Values, tolerance);
+
+		public ExpectedPixels(string name, float x, float y, Color[,] pixels, byte tolerance = 0)
+		{
+			Name = name;
+			X = (int)x;
+			Y = (int)y;
+			Values = pixels;
+			Tolerance = tolerance;
+		}
+
+		public ExpectedPixels(string name, float x, float y, string[,] pixels, byte tolerance = 0)
+		{
+			var colors = new Color[pixels.GetLength(0), pixels.GetLength(1)];
+			for (var py = 0; py < pixels.GetLength(0); py++)
+			for (var px = 0; px < pixels.GetLength(1); px++)
+			{
+				var colorCode = pixels[py, px];
+				colors[py, px] = string.IsNullOrWhiteSpace(colorCode)
+					? Color.Empty
+					: ColorCodeParser.Parse(colorCode);
+			}
+
+			Name = name;
+			X = (int)x;
+			Y = (int)y;
+			Values = colors;
+			Tolerance = tolerance;
+		}
+
+		public string Name { get; set; }
+		public int X { get; }
+		public int Y { get; }
+		public Color[,] Values { get; }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this multidimensional array to a jagged array.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4919453)